core/common/src/Utf8.kt

```diff
@@ -118,445 +117,415 @@ fun String.utf8Size(beginIndex: Int = 0, endIndex: Int = length): Long {
   return result
 }
 
-internal const val REPLACEMENT_BYTE: Byte = '?'.code.toByte()
-internal const val REPLACEMENT_CHARACTER: Char = '\ufffd'
-internal const val REPLACEMENT_CODE_POINT: Int = REPLACEMENT_CHARACTER.code
+/**
+ * Encodes [codePoint] in UTF-8 and writes it to this sink.
+ *
+ * @param codePoint the codePoint to be written.
+ *
+ * @throws IllegalStateException when the sink is closed.
+ */
+@OptIn(DelicateIoApi::class)
+public fun Sink.writeUtf8CodePoint(codePoint: Int): Unit =
```

Comment:
I'd suggest hiding all code-point signatures as well.

We plan adding a bit more nuanced code point support to multiplatform (https://youtrack.jetbrains.com/issue/KT-30509), would be a shame to have all the cool names taken

Reply:
Done

Reply:
`writeUtf8CodePoint` is actively used in Json integration (to do e.g. `write(":")`), so I suggest making it public now.